tools/rimage/config/mt8196.toml

```diff
@@ -0,0 +1,15 @@
+version = [1, 0]  # use simple file write
+
+[adsp]
+name = "mt8196"
+
+[[adsp.mem_zone]]
+type = "IRAM"
+base = "0x4e100000"
+size = "0x00080000"
+host_offset = "0x0"
+[[adsp.mem_zone]]
+type = "SRAM"
+base = "0x90000000"
+size = "0x00600000"
+host_offset = "0x0"
```

Comment:
in fact we can now preprocess toml with gcc, as we do for Intal ADSP platforms, so getting values from DTS might be possible?

Reply:
Yeah.  Though this particular block (not all the module config) is almost entirely generatable.   I guess my gut says what we want is to construct and emit it ourselves from cmake (or a script invoked from there) instead of trying to do it with the preprocessor.  But that takes some thought and design.

I've made the point in the past, but SOF seems to have a love affair with weak programming languages (cpp here, also m4, the topology2 changes to alsaconf, frankly alsaconf itself seems to have always been too limited pre-sof, alsabat is sort of in that list too) and then getting in trouble when the requirements expand farther than the syntax can handle.   There's no shame in using python from the start. :)